### PR TITLE
feat: onboarding wizard

### PR DESCRIPTION
**Pull request by christophenne**

### CODE REVIEW

File: frontend/cloud-ui/src/app/components/onboarding-wizard/onboarding-wizard.component.html [hunk 1]
```diff
@@ -0,0 +1,287 @@
+<div
+  @modalFlyInOut
+  style="transform-origin: top center"
+  class="p-4 w-full mt-12 max-h-full bg-white rounded-lg shadow dark:bg-gray-900">
+  <!-- Modal header -->
+  <div class="flex items-center justify-between p-4 md:p-5 border-b rounded-t dark:border-gray-600">
+    <h3 class="text-xl font-semibold text-gray-900 dark:text-white">Let's get you started</h3>
```
Comment (pmig) [inline frontend/cloud-ui/src/app/components/onboarding-wizard/onboarding-wizard.component.html]:
```suggestion
    <h2 class="text-xl font-semibold text-gray-900 dark:text-white">Onboard your first customer</h2>
```

File: internal/handlers/applications.go [hunk 2]
```diff
@@ -242,6 +243,41 @@ func getApplicationVersionComposeFile(w http.ResponseWriter, r *http.Request) {
...
+	if err := db.CreateApplication(ctx, &application); err != nil {
+		log.Warn("could not create sample application", zap.Error(err))
+		w.WriteHeader(http.StatusInternalServerError)
+		if _, err = fmt.Fprintln(w, err); err != nil {
+			log.Error("failed to write error to response", zap.Error(err))
+		}
+		return
+	}
+	version := types.ApplicationVersion{
+		Name:            "v1.0.0",
```
Comment (pmig) [inline internal/handlers/applications.go]:
```suggestion
		Name:            "v1.7.1",
```

Comment (pmig) [reply internal/handlers/applications.go]: we could also add a renovate automation here and in the docker-compose file :D

File: internal/handlers/applications.go [hunk 3]
```diff
@@ -258,3 +294,47 @@ func applicationMiddleware(next http.Handler) http.Handler {
...
 }
+
+// TODO maybe shiori is not the best sample app, because it doesn't have a "production" docker compose file
+// from https://github.com/go-shiori/shiori/blob/master/docker-compose.yaml
+var shioriComposeFile = []byte(`# Docker compose for development purposes only.
+# Edit it to fit your current development needs.
+version: "3"
+services:
+  shiori:
+    build:
```
Comment (pmig) [inline internal/handlers/applications.go]: as discussed we should use an image here

File: internal/handlers/applications.go [hunk 4]
```diff
@@ -258,3 +294,47 @@ func applicationMiddleware(next http.Handler) http.Handler {
...
+    links:
+      - "postgres"
+      - "mariadb"
+    environment:
+      SHIORI_DIR: /srv/shiori
+      #SHIORI_DATABASE_URL: mysql://shiori:shiori@(mariadb)/shiori?charset=utf8mb4
+      SHIORI_DATABASE_URL: postgres://shiori:shiori@postgres/shiori?sslmode=disable
+
+  postgres:
+    image: postgres:15
```
Comment (pmig) [inline internal/handlers/applications.go]:
```suggestion
    image: postgres:17.2-alpine3.20
```

File: internal/handlers/applications.go [hunk 5]
```diff
@@ -258,3 +294,47 @@ func applicationMiddleware(next http.Handler) http.Handler {
...
+
+  mariadb:
+    image: mariadb:11
+    environment:
+      MYSQL_ROOT_PASSWORD: toor
+      MYSQL_DATABASE: shiori
+      MYSQL_USER: shiori
+      MYSQL_PASSWORD: shiori
+    ports:
+      - "3306:3306"
```
Comment (pmig) [inline internal/handlers/applications.go]:
```suggestion

```

File: frontend/cloud-ui/src/app/components/onboarding-wizard/onboarding-wizard-stepper.component.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  isApplicationStep(): boolean {
+    return this.selectedIndex == 0;
+  }
+
+  isDeploymentTargetStep(): boolean {
+    return this.selectedIndex == 1;
+  }
+
+  isFinalStep(): boolean {
+    return this.selectedIndex == 2;
```
Comment (kosmoz) [inline frontend/cloud-ui/src/app/components/onboarding-wizard/onboarding-wizard-stepper.component.ts]:
```suggestion
    return this.selectedIndex === this.steps.length - 1;
```

File: frontend/cloud-ui/src/app/components/onboarding-wizard/onboarding-wizard-stepper.component.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  currentFormGroup() {
+    return this.selected!.stepControl as FormGroup;
+  }
+
+  isApplicationStep(): boolean {
+    return this.selectedIndex == 0;
+  }
+
+  isDeploymentTargetStep(): boolean {
+    return this.selectedIndex == 1;
```
Comment (kosmoz) [inline frontend/cloud-ui/src/app/components/onboarding-wizard/onboarding-wizard-stepper.component.ts]:
```suggestion
    return this.selectedIndex === 1;
```

File: frontend/cloud-ui/src/app/components/onboarding-wizard/onboarding-wizard-stepper.component.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  protected readonly shipIcon = faShip;
+  protected readonly networkIcon = faNetworkWired;
+  @Output('attemptContinue') attemptContinueOutput: EventEmitter<void> = new EventEmitter();
+
+  currentFormGroup() {
+    return this.selected!.stepControl as FormGroup;
+  }
+
+  isApplicationStep(): boolean {
+    return this.selectedIndex == 0;
```
Comment (kosmoz) [inline frontend/cloud-ui/src/app/components/onboarding-wizard/onboarding-wizard-stepper.component.ts]:
```suggestion
    return this.selectedIndex === 0;
```

File: frontend/cloud-ui/src/app/components/onboarding-wizard/onboarding-wizard.component.ts [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+  }
+
+  private destroyed$: Subject<void> = new Subject();
+  ngOnDestroy() {
+    this.destroyed$.complete();
+  }
+
+  onFileSelected(event: any) {
+    this.fileToUpload = event.target.files[0];
+  }
```
Comment (kosmoz) [inline frontend/cloud-ui/src/app/components/onboarding-wizard/onboarding-wizard.component.ts]:
```suggestion
  onFileSelected(event: Event) {
    this.fileToUpload = (event.target as HTMLInputElement).files?.[0] ?? null;
  }
```

File: frontend/cloud-ui/src/app/components/onboarding-wizard/onboarding-wizard.component.ts [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+
+  attemptContinue() {
+    if (this.loading) {
+      return;
+    }
+
+    if (this.stepper.selectedIndex == 0) {
```
Comment (kosmoz) [inline frontend/cloud-ui/src/app/components/onboarding-wizard/onboarding-wizard.component.ts]:
```suggestion
    if (this.stepper.selectedIndex === 0) {
```

File: frontend/cloud-ui/src/app/components/onboarding-wizard/onboarding-wizard.component.ts [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+              ),
+              withLatestFrom(this.applications.list())
+            )
+            .subscribe(([version, apps]) => {
+              this.createdApp = apps.find((a) => a.id === version.applicationId);
+              this.nextStep();
+            });
+        }
+      }
+    } else if (this.stepper.selectedIndex == 1) {
```
Comment (kosmoz) [inline frontend/cloud-ui/src/app/components/onboarding-wizard/onboarding-wizard.component.ts]:
```suggestion
    } else if (this.stepper.selectedIndex === 1) {
```

File: frontend/cloud-ui/src/app/components/onboarding-wizard/onboarding-wizard.component.ts [hunk 12]
```diff
@@ -0,0 +1,210 @@
...
+      if (this.deploymentTargetForm.valid) {
+        this.loading = true;
+        const base = {
+          name: this.deploymentTargetForm.controls.customerName.value!.toLowerCase().replaceAll(' ', '-'),
```
Comment (kosmoz) [inline frontend/cloud-ui/src/app/components/onboarding-wizard/onboarding-wizard.component.ts]: As discussed, IMO this transformation should be removed for now
```suggestion
          name: this.deploymentTargetForm.controls.customerName.value!,
```

File: frontend/cloud-ui/src/app/components/onboarding-wizard/onboarding-wizard.component.ts [hunk 13]
```diff
@@ -0,0 +1,210 @@
...
+          type: 'docker',
+          geolocation: {
+            lat: 48.1956026,
+            lon: 16.3633028,
+          },
+        };
+        this.deploymentTargets
+          .create({
+            ...base,
+            name: base.name + '-staging',
```
Comment (kosmoz) [inline frontend/cloud-ui/src/app/components/onboarding-wizard/onboarding-wizard.component.ts]: Suggestion:
```suggestion
            name: base.name + ' (staging)',
```

File: frontend/cloud-ui/src/app/components/onboarding-wizard/onboarding-wizard.component.ts [hunk 14]
```diff
@@ -0,0 +1,210 @@
...
+                applicationVersionId: this.createdApp!.versions![0].id!,
+                deploymentTargetId: dt.id!,
+              })
+            )
+          )
+          .subscribe(() => this.nextStep());
+      }
+    } else if (this.stepper.selectedIndex == 2) {
+      // TODO
+      window.location.href = '/dashboard';
```
Comment (kosmoz) [inline frontend/cloud-ui/src/app/components/onboarding-wizard/onboarding-wizard.component.ts]: should use `router.navigate(['/dashboard'])` but we can do this later aas well

**Pull request merged**